mod_modular_vanilla/hooks/config/character.nut

```diff
@@ -11,7 +16,35 @@
 	MV_DiversionDamageMult = 0.75
 });
 
+local original_getClone = ::Const.CharacterProperties.getClone;
 ::MSU.Table.merge(::Const.CharacterProperties, {
+	function getClone()
+	{
+		local ret = original_getClone();
+		ret.MV_MoraleCheckBraveryCallbacks = clone this.MV_MoraleCheckBraveryCallbacks;
+		ret.MV_ForbiddenMoraleStates = clone this.MV_ForbiddenMoraleStates;
+		return ret;
+	},
+
 	// Part of modularization of player_party.updateStrength
 	MV_StrengthMult = 1.0
+	/*
+	You push functions to this array during `skill.onUpdate` or `skill.onAfterUpdate`.
+	These are then used to modify the Bravery during actor.checkMorale.
+	The functions pushed look like this:
+		function( _change, _type )
+		_change is the _change parameter in `actor.checkMorale`.
+		_type is the _type parameter in `actor.checkMorale`.
+	The functions return a table with this signature:
+	{
+		Add = <integer>
+		Mult = <float>
+	}
+	// Add is then added to Bravery during actor.checkMorale
+	// whereas Mult is multiplied with the BraveryMult.
+	*/
+	MV_MoraleCheckBraveryCallbacks = [],
+	// During actor.setMoraleState if the passed morale state is found in this
+	// array, the morale state will not be set to that value..
+	MV_ForbiddenMoraleStates = []
```

Comment:
Do we want this or do we want two properties `MV_MaxMoraleState` and `MV_MinMoraleState` because in the context where this thing is used is basically relevant for that only i.e. during a morale check we check if the actor can go to that morale state or not. Although there is a `actor.m.MaxMoraleState` in vanilla and if we implement a property for that then we'd have to think about how to tie that in.

Reply:
Or just return false from callback. Will be easy to make this dynamic - say cannot flee if you full hp or something.

Reply:
There is no callback here. In this particular case it is simple because skills can push things to it during their `onUpdate` function. So it is easy to make dynamic already e.g. skill in its `onUpdate` checks HP and then decides to push a certain morale state as forbidden state.

The main question is do we need an array of forbidden states? Or do we just need to know the min/max morale states? Because in practicality all these "forbidden states" do is just to prevent state from being increased or reduced to that state. We don't actually ever "jump over" morale states.